### PR TITLE
Replace UUIDv4 with ULID [SFEQS-835]

### PR DESCRIPTION
**Pull request by lucacavallaro**
#### List of Changes

The `Id` value object now is based on `ULID` instead of `UUIDv4`.  The rationale behind this decision is described in `ADR #3`.

#### How Has This Been Tested?

The changes were tested locally.

#### Types of changes

- [x] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: docs/architecture/decisions/0003-choose-ulid-as-identifier-format.md
Comment (gunzip) [inline docs/architecture/decisions/0003-choose-ulid-as-identifier-format.md]:
```suggestion
Accepted
```

**Review by gunzip — APPROVED** [review]

**Review by grausof — APPROVED** [review]
OK!

File: docs/architecture/decisions/0003-choose-ulid-as-identifier-format.md
Comment (lucacavallaro) [reply docs/architecture/decisions/0003-choose-ulid-as-identifier-format.md]: I was waiting for your approval!

**Pull request merged**
